### ISSUE TITLE
Asset Pipeline

### ISSUE THREAD

**Issue by niallobrien** [adonisjs/discussion#24]
I think having a default asset pipeline utilising Webpack would be a great addition to the framework. We can already see other frameworks moving towards this direction as both Laravel and Ruby on Rails are/will be moving their asset pipelines to Webpack.
I know @RomainLanz already utilises Laravel Mix with his Adonis projects. Perhaps this is something that could be made official with a few Adonis specifics added on top (eg. loading hashed/un-hashed assets based on environment etc.) where needed.

Thoughts?

**Comment by RomainLanz** [comment]
Hey !

I think that it isn't a good idea to have a default assets pipeline in the Framework. If Adonis start to do that people will complain about "why not using gulp", "why not using grunt", "why not using X".

This will only add work to do. Instead we (community) can write tutorial to teach how to use one type of pipeline.

As you said I use `Laravel Mix` so I could write an article about how to setup `Laravel Mix` with Adonis (which take ~5 minutes) if it is necessary.

**Comment by niallobrien** [comment]
Yeah, I know what you mean. Okay, maybe not a full integration but if Adonis were to make some recommendations and maybe recommended Mix for most scenarios for example, the docs could have a 5 minute guide like you said.

**Comment by bitkidd** [comment]
I think that each developer should make this decision by himself, as soon as there are too many different cases and plenty of instruments for each of them.

**Comment by thetutlage** [comment]
Frontend is a weird eco-system and people have very specific use-cases. I have seen developers not even using the default `webpack` file created by Vue-cli on their new projects.

Getting into this will divert my attention from what matters the most. I am not adding any assets pipeline ever

**Comment by thetutlage** [comment]
Webpack, webpack and webpack. There is no point in re-discovering what's out there

**Comment by pedzed** [comment]
> I am not adding any assets pipeline ever

I think this is a bit stubborn. [Laravel Mix](https://github.com/JeffreyWay/laravel-mix) is the perfect fit for AdonisJS, especially considering that Adonis is majorly inspired by Laravel. And Laravel is using it too, so why not do the same?

When I chose AdonisJS, I chose it because I wanted a full-fledged JavaScript framework. Not including Laravel Mix or something similar is to me a house with a backyard that has not been taken care of.

I vote for reopening the issue and reconsidering the options.
